conans/client/cmake.py

```diff
@@ -115,18 +118,24 @@ def flags(self):
         op_system = str(self._settings.os) if self._settings.os else None
         arch = str(self._settings.arch) if self._settings.arch else None
         comp = str(self._settings.compiler) if self._settings.compiler else None
-        comp_version = self._settings.compiler.version
+        comp_version = self._settings.compiler.version     
+
+        try:
+            op_system_version = str(self._settings.os.version) if self._settings.os.version else None
```

Comment:
Sure, sure, it is fine, guarded with the try-except. Just that it doesn't match the definition of current settings (put otherwise, it will always ``raise``, except if you customize your settings.yml). Also the use case seems to require adding other ``os`` as WindowsCE. 

As long as you are using it with your own ``settings.yml`` definition defining this ``os.version``, as it is working for you, I am fine with it, no need to change anything. Just put some comment in the code, maybe with a snippet of a example working ``settings.yml`` (not the full file, just the relevant).

If you can't fix the tests by Friday, don't worry at all, we will fix them, so at least there is some time during the weekend for testing before the release. Thanks very much!